pkg/sanity/groupcontroller.go

```diff
@@ -0,0 +1,341 @@
+			Skip("GroupControllerService not supported")
+		}
+
+		r = &Resources{
+			Context:               sc,
+			ControllerClient:      csi.NewControllerClient(sc.ControllerConn),
+			GroupControllerClient: csi.NewGroupControllerClient(sc.ControllerConn),
+			NodeClient:            csi.NewNodeClient(sc.Conn),
+		}
+	})
+
+	AfterEach(func() {
+		r.Cleanup()
+	})
+
+	Describe("GroupControllerGetCapabilities", func() {
+		It("should return appropriate capabilities", func() {
+			caps, err := r.GroupControllerGetCapabilities(
+				context.Background(),
+				&csi.GroupControllerGetCapabilitiesRequest{})
+
+			By("checking successful response")
+			Expect(err).NotTo(HaveOccurred())
+			Expect(caps).NotTo(BeNil())
+			Expect(caps.GetCapabilities()).NotTo(BeNil())
+
+			for _, cap := range caps.GetCapabilities() {
+				Expect(cap.GetRpc()).NotTo(BeNil())
+
+				switch cap.GetRpc().GetType() {
+				case csi.GroupControllerServiceCapability_RPC_CREATE_DELETE_GET_VOLUME_GROUP_SNAPSHOT:
+				default:
+					Fail(fmt.Sprintf("Unknown capability: %v\n", cap.GetRpc().GetType()))
+				}
+			}
+		})
+	})
+})
+
+var _ = DescribeSanity("GroupController Service [GroupController VolumeGroupSnapshots]", func(sc *TestContext) {
+	var r *Resources
+
+	BeforeEach(func() {
+		var providesGroupControllerService bool
+
+		i := csi.NewIdentityClient(sc.Conn)
+		req := &csi.GetPluginCapabilitiesRequest{}
+		res, err := i.GetPluginCapabilities(context.Background(), req)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(res).NotTo(BeNil())
+
+		for _, cap := range res.GetCapabilities() {
+			switch cap.GetType().(type) {
+			case *csi.PluginCapability_Service_:
+				switch cap.GetService().GetType() {
+				case csi.PluginCapability_Service_GROUP_CONTROLLER_SERVICE:
+					providesGroupControllerService = true
+				}
+			}
+		}
+
+		if !providesGroupControllerService {
+			Skip("GroupControllerService not supported")
+		}
+
+		r = &Resources{
+			Context:               sc,
+			ControllerClient:      csi.NewControllerClient(sc.ControllerConn),
+			GroupControllerClient: csi.NewGroupControllerClient(sc.ControllerConn),
+			NodeClient:            csi.NewNodeClient(sc.Conn),
+		}
+
+		if !isGroupControllerCapabilitySupported(r, csi.GroupControllerServiceCapability_RPC_CREATE_DELETE_GET_VOLUME_GROUP_SNAPSHOT) {
+			Skip("VolumeGroupSnapshots not supported")
+		}
+	})
+
+	AfterEach(func() {
+		r.Cleanup()
+	})
+
+	Describe("CreateVolumeGroupSnapshot", func() {
+		It("should fail when no name is provided", func() {
+			_, err := r.CreateVolumeGroupSnapshot(
+				context.Background(),
+				&csi.CreateVolumeGroupSnapshotRequest{
+					Secrets: sc.Secrets.CreateSnapshotSecret,
+				},
+			)
+			Expect(err).To(HaveOccurred())
+
+			serverError, ok := status.FromError(err)
+			Expect(ok).To(BeTrue())
+			Expect(serverError.Code()).To(Equal(codes.InvalidArgument), "unexpected error: %s", serverError.Message())
+		})
+
+		/* disabled tests, need modifications for VolumeGroupSnapshot
```

Comment:
Are you planning to add more tests in this PR or in followup PRs?
If you want to get this PR merged first and add followup PRs later, please delete the commented out code and add a TODO for adding followup tests.

Reply:
I have file #470 for this now.